presidio-analyzer/analyzer/entity_recognizer.py

```diff
@@ -132,28 +136,34 @@ def enhance_using_context(self, text, raw_results,
                 word=text[result.start:result.end],
                 start=result.start)
 
-            context_similarity = self.__calculate_context_similarity(
+            supportive_context_word = self.__find_supportive_context_word(
                 context, predefined_context_words)
-            if context_similarity >= \
-               self.CONTEXT_SIMILARITY_THRESHOLD:
+            if supportive_context_word != "":
                 result.score += \
-                  context_similarity * self.CONTEXT_SIMILARITY_FACTOR
+                  self.CONTEXT_SIMILARITY_FACTOR
                 result.score = max(
                     result.score,
                     self.MIN_SCORE_WITH_CONTEXT_SIMILARITY)
                 result.score = min(
                     result.score,
                     EntityRecognizer.MAX_SCORE)
+
+                # Update the explainability object with context information
+                # helped improving the score
+                result.analysis_explanation.set_supportive_context_word(
+                    supportive_context_word)
+                result.analysis_explanation.set_improved_score(result.score)
```

Comment:
Here we should add the call for .set_pattern_verification_output (checksum passed yes or no)

Reply:
But you wanted the result of the checksum, not only if it passed. Am I wrong?

Reply:
Yes. In many cases it's the same, but providing the full checksum result is preferred.